tests/test_parser_stress.py

```diff
@@ -0,0 +1,350 @@
+    return all(n % i for i in range(3, int(math.sqrt(n)) + 1, 2))
+
+
+def is_power_of(x, base):
+    n = x
+    while(n > 0):
+        if n == 1:
+            return True
+        if n % base:
+            return False
+        n /= base
+    raise Exception('is_power_of: unexpected result with x = ' +
+                    str(x) + ' and base = ' + str(base))
+
+
+def is_square(n):
+    sq = int(math.sqrt(n))
+    return sq * sq == n
+
+
+def is_multiple(n, base):
+    return n % base == 0
+
+
+def is_interesting_number(n):
+    if n < 1:
+        return False
+
+    if n <= 1000 and is_prime(n):
+        return True
+
+    if is_power_of(n, 2) \
+            or is_power_of(n-1, 2) \
+            or is_power_of(n+1, 2):
+        return True
+
+    if is_power_of(n, 10) \
+            or is_power_of(n-1, 10) \
+            or is_power_of(n+1, 10):
+        return True
+
+    if is_multiple(n, 1024) \
+            or is_multiple(n-1, 1024) \
+            or is_multiple(n+1, 1024):
+        return True
+
+    if is_multiple(n, 1000) \
+            or is_multiple(n-1, 1000) \
+            or is_multiple(n+1, 1000):
+        return True
+
+    if is_square(n):
+        return True
+
+    if n <= 64:
+        return True
+
+    return False
+
+
+def is_interesting_number_short(n):
+    if n < 1:
+        return False
+
+    if n <= 100 and is_prime(n):
+        return True
+
+    if is_power_of(n, 2) \
+            or is_power_of(n-1, 2) \
+            or is_power_of(n+1, 2):
+        return True
+
+    if is_power_of(n, 10) \
+            or is_power_of(n-1, 10) \
+            or is_power_of(n+1, 10):
+        return True
+
+    if is_multiple(n, 4 * 1024) \
+            or is_multiple(n-1, 1024) \
+            or is_multiple(n+1, 1024):
+        return True
+
+    if is_multiple(n, 5 * 1000) \
+            or is_multiple(n-1, 1000) \
+            or is_multiple(n+1, 1000):
+        return True
+
+    if is_square(n) and n <= 100:
+        return True
+
+    if n <= 32:
+        return True
+
+    return False
+
+
+def get_max_number():
+    return 70 * 1000 + 1
+
+
+def get_interesting_numbers(short=False):
+    if short:
+        interesting_numbers = (x for x in range(1, 17000)
+                               if is_interesting_number_short(x))
+    else:
+        interesting_numbers = (x for x in range(1, get_max_number() + 1)
+                               if is_interesting_number(x))
+    return list(interesting_numbers)
+
+
+class TestingValueTarget(ValueTarget):
+    def __init__(self):
+        super().__init__()
+        self.started = False
+        self.finished = False
+
+    def start(self):
+        self.started = True
+        super().start()
+
+    def finish(self):
+        self.finished = True
+        super().finish()
+
+
+class DifferentChunksTestCase(TestCase):
+
+    def test_basic_last_attach(self):
+        data = get_random_bytes(1024 * 1024, 159)
+        self.do_test(data, 'random_bytes', True)
+        pass
```

Comment:
This `pass` statement is not required.

Reply:
those pass statements were put for easy commenting of particular test sets. I will remove them